### PR TITLE
Bump opentelemetry-go versions to latest stable/unstable

### PR DESCRIPTION
**Pull request by lrascao**

### CODE REVIEW

File: exporter/metric/metric.go [hunk 1]
```diff
@@ -126,6 +128,27 @@ func (me *metricExporter) Export(ctx context.Context, rm metricdata.ResourceMetr
 	)
 }
 
+// Temporality returns the Temporality to use for an instrument kind.
+func (me *metricExporter) Temporality(k metric.InstrumentKind) metricdata.Temporality {
```
Comment (lrascao) [inline exporter/metric/metric.go:132]: both `Temporality` and `Aggregator` are two new interface methods and this is the bit i need some guidance on how they should work with gcp

Comment (dashpole) [reply exporter/metric/metric.go:132]: We should use cumulative temporality and the default aggregators.  Those should be the same as the OTLP exporter's defaults.

Comment (lrascao) [reply exporter/metric/metric.go:132]: should the temporality not depend on the instrument kind? felt like Cumulative would fit best gauges and Delta fit counters

**Comment by lrascao** [comment]
```
Step #1 - "run-tests-local": === RUN   TestBasicTrace
Step #1 - "run-tests-local":     trace_test.go:80: Retrying GetTrace: googleapi: Error 404: Requested entity was not found., notFound
Step #1 - "run-tests-local":     trace_test.go:80: Retrying GetTrace: googleapi: Error 404: Requested entity was not found., notFound
Step #1 - "run-tests-local":     trace_test.go:80: Retrying GetTrace: googleapi: Error 404: Requested entity was not found., notFound
Step #1 - "run-tests-local":     trace_test.go:80: Retrying GetTrace: googleapi: Error 404: Requested entity was not found., notFound
Step #1 - "run-tests-local":     trace_test.go:80: Retrying GetTrace: googleapi: Error 404: Requested entity was not found., notFound
Step #1 - "run-tests-local":     trace_test.go:80: Retrying GetTrace: googleapi: Error 404: Requested entity was not found., notFound
Step #1 - "run-tests-local":     trace_test.go:80: Retrying GetTrace: googleapi: Error 404: Requested entity was not found., notFound
Step #1 - "run-tests-local":     trace_test.go:80: Retrying GetTrace: googleapi: Error 404: Requested entity was not found., notFound
Step #1 - "run-tests-local":     trace_test.go:80: Retrying GetTrace: googleapi: Error 404: Requested entity was not found., notFound
Step #1 - "run-tests-local":     trace_test.go:80: Retrying GetTrace: googleapi: Error 404: Requested entity was not found., notFound
Step #1 - "run-tests-local":     trace_test.go:80: Retrying GetTrace: googleapi: Error 404: Requested entity was not found., notFound
Step #1 - "run-tests-local":     trace_test.go:85: 
Step #1 - "run-tests-local":         	Error Trace:	trace_test.go:85
Step #1 - "run-tests-local":         	            				trace_test.go:106
Step #1 - "run-tests-local":         	Error:      	Received unexpected error:
Step #1 - "run-tests-local":         	            	retryable: googleapi: Error 404: Requested entity was not found., notFound
Step #1 - "run-tests-local":         	Test:       	TestBasicTrace
Step #1 - "run-tests-local": --- FAIL: TestBasicTrace (13.43s)
```

hm... maybe the terraform step didn't fully complete?

File: exporter/metric/metric.go (hunk 1 above)
Comment (dashpole) [reply exporter/metric/metric.go]: GCP currently uses cumulative counters.

File: Makefile [hunk 2]
```diff
@@ -196,7 +196,6 @@ update-otel:
 	$(MAKE) update-dep MODULE=go.opentelemetry.io/contrib/detectors/gcp VERSION=$(STABLE_CONTRIB_OTEL_VERSION)
 	$(MAKE) update-dep MODULE=go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp VERSION=$(UNSTABLE_CONTRIB_OTEL_VERSION)
 	$(MAKE) update-dep MODULE=go.opentelemetry.io/collector VERSION=$(COLLECTOR_VERSION)
-	$(MAKE) update-dep MODULE=go.opentelemetry.io/collector/pdata VERSION=$(COLLECTOR_VERSION)
```
Comment (dashpole) [inline Makefile:199]: why is this removed?

**Comment by dashpole** [comment]
> hm... maybe the terraform step didn't fully complete?

retried

File: Makefile (hunk 2 above)
Comment (lrascao) [reply Makefile:199]: `pdata` made it to `v1.0.0` on the `v0.67.0` [release](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.67.0)

Comment (dashpole) [reply Makefile:199]: Got it.  Can you split it into stable collector version and unstable collector version instead of removing?

**Comment by lrascao** [comment]
> /gcbrun

Guess this needs another retry?

**Comment by zchee** [comment]
@damemi @dashpole 
FYI, CircleCI said,

> Could not find a usable config.yml, you may have revoked the CircleCI OAuth app.
Please sign out of CircleCI and log back in with your VCS before triggering a new pipeline.

**Comment by damemi** [comment]
Weird, I can't re-run it circleci either. Going to try closing and reopening

**Comment by lrascao** [comment]
@damemi I've revoked and reinstated my GitHub circleCI access, can you retry please?

**Comment by dashpole** [comment]
```
golangci-lint in ./exporter/collector/integrationtest
config/config_test.go:37:20: SA1019: componenttest.NopFactories is deprecated: [v0.68.0] use otelcoltest.NopFactories. (staticcheck)
	factories, err := componenttest.NopFactories()
	                  ^
config/config_test.go:56:22: SA1019: config.NewExporterSettings is deprecated: [v0.68.0] will be removed soon, Config no longer requires to embed this. (staticcheck)
			ExporterSettings: config.NewExporterSettings(component.NewIDWithName(typeStr, "customname")),
			                  ^
config/config_test.go:114:21: SA1019: config.NewExporterSettings is deprecated: [v0.68.0] will be removed soon, Config no longer requires to embed this. (staticcheck)
		ExporterSettings: config.NewExporterSettings(component.NewID(typeStr)),
		                  ^
```

File: exporter/metric/metric.go [hunk 3]
```diff
@@ -126,6 +128,16 @@ func (me *metricExporter) Export(ctx context.Context, rm metricdata.ResourceMetr
 	)
 }
 
+// Temporality returns the Temporality to use for an instrument kind.
+func (me *metricExporter) Temporality(ik metric.InstrumentKind) metricdata.Temporality {
+	return metric.DefaultTemporalitySelector(ik)
```
Comment (zchee) [inline exporter/metric/metric.go:133]: How about implementing option like `stdoutmetric`?

- https://github.com/open-telemetry/opentelemetry-go/blob/exporters/stdout/stdoutmetric/v0.34.0/exporters/stdout/stdoutmetric/config.go#L78-L127

**Comment by lrascao** [comment]
CircleCI all green, ready for some gcbrun @dashpole

**Comment by lrascao** [comment]
`-gce": === RUN   TestBasicTrace
Step #1 - "run-tests-gce":     trace_test.go:80: Retrying GetTrace: googleapi: Error 404: Requested entity was not found., notFound
Step #1 - "run-tests-gce": === RUN   TestBasicTrace/Span_has_label_g.co/agent
Step #1 - "run-tests-gce": === RUN   TestBasicTrace/Span_has_label_test_id`

looks like it needs a retry

**Comment by dashpole** [comment]
(please ignore cloud run and gae tests.  they are under development)

**Comment by lrascao** [comment]
So we'll wait until the tests become stable to unblock?

File: exporter/metric/metric.go (hunk 3 above)
Comment (dashpole) [reply exporter/metric/metric.go:133]: Our exporter currently assumes we only get cumulative temporality metrics: https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/main/exporter/metric/metric.go#L347.  I don't believe Cloud Monitoring doesn't allow delta for custom metrics: https://cloud.google.com/monitoring/api/v3/kinds-and-types#kind-type-combos.  So we should stick with only the default for now.

We can add this in a follow-up if it turns out to be needed.

**Review by dashpole — APPROVED** [review]

**Comment by dashpole** [comment]
Thanks for the contribution @lrascao!

File: exporter/metric/metric.go (hunk 3 above)
Comment (zchee) [reply exporter/metric/metric.go:133]: @dashpole I'm totally understand now about Cloud Monitoring temporality metrics spec, make sense.

From your description, I have another concern about use `DefaultTemporalitySelector` might be confusing a bit because not included `CumulativeTemporality` word. But it seems related to otel sdk implementation opinions.

- https://github.com/open-telemetry/opentelemetry-go/blob/sdk/metric/v0.34.0/sdk/metric/reader.go#L115-L117

> We can add this in a follow-up if it turns out to be needed.

IMO, would be adding "note" comment to GoogleCloudPlatform/opentelemetry-operations-go side. But very nits.
Thanks
